error_stats/geolocation_error_stats.py

```diff
@@ -0,0 +1,542 @@
+3) Scale to nadir-equivalent using geometric factors
+4) Compute statistical performance metrics
+"""
+
+import numpy as np
+import xarray as xr
+from typing import Dict, Optional, Tuple, Union
+from dataclasses import dataclass
+
+
+@dataclass
+class GeolocationConfig:
+    """Configuration parameters for geolocation processing."""
+    earth_radius_m: float = 6378140.0
+    performance_threshold_m: float = 250.0
+    performance_spec_percent: float = 39.0
+
+
+class ErrorStatsProcessor:
+    """Production-ready processor for geolocation error statistics."""
+
+    def __init__(self, config: Optional[GeolocationConfig] = None):
+        """Initialize processor with configuration."""
+        self.config = config or GeolocationConfig()
+
+    def process_geolocation_errors(self, input_data: xr.Dataset) -> xr.Dataset:
+        """
+        Process geolocation errors from input dataset to nadir-equivalent statistics.
+
+        Args:
+            input_data: Xarray Dataset with required error measurement variables
+
+        Returns:
+            Xarray Dataset with processed results and statistics
+        """
+        # Validate input data
+        self._validate_input_data(input_data)
+
+        # Extract data arrays
+        n_measurements = len(input_data.measurement)
+
+        # Convert angular errors to distance errors
+        lat_error_rad = np.deg2rad(input_data.lat_error_deg.values)
+        lon_error_rad = np.deg2rad(input_data.lon_error_deg.values)
+        cp_lat_rad = np.deg2rad(input_data.cp_lat_deg.values)
+        cp_lon_rad = np.deg2rad(input_data.cp_lon_deg.values)
+
+        # Calculate N-S and E-W error distances in meters
+        ns_error_dist_m = self.config.earth_radius_m * lat_error_rad
+        ew_error_dist_m = self.config.earth_radius_m * np.cos(cp_lat_rad) * lon_error_rad
+
+        # Transform boresight vectors from HS to CTRS coordinates
+        bhat_ctrs = self._transform_boresight_vectors(
+            input_data.bhat_hs.values,
+            input_data.t_hs2ctrs.values
+        )
+
+        # Process each measurement to nadir-equivalent
+        results = self._process_to_nadir_equivalent(
+            ns_error_dist_m, ew_error_dist_m,
+            input_data.riss_ctrs.values, bhat_ctrs,
+            cp_lat_rad, cp_lon_rad, n_measurements
+        )
+
+        # Create output dataset
+        output_data = self._create_output_dataset(input_data, results)
+
+        # Add statistics as global attributes
+        stats = self._calculate_statistics(results['nadir_equiv_total_error_m'])
+        output_data.attrs.update(stats)
+
+        return output_data
+
+    def _validate_input_data(self, data: xr.Dataset) -> None:
+        """Validate that input dataset contains all required variables."""
+        required_vars = [
+            'lat_error_deg', 'lon_error_deg', 'riss_ctrs', 'bhat_hs',
+            't_hs2ctrs', 'cp_lat_deg', 'cp_lon_deg', 'cp_alt'
+        ]
```

Comment:
This seems very mission-specific. Does this really belong in a generic geolocation library?

Reply:
Adding a comment here to remind myself. I'm going to be pulling out mission-specific code during further development and integration of this with the GCS module - plan for now is that these variables will be named in a config json.